app/knights_config.py

```diff
@@ -0,0 +1,121 @@
+from app.knights import Knight
+from app.ammunition.weapon import Weapon
+from app.ammunition.armour import Armour
+from app.ammunition.potion import Potion
+
+KNIGHTS = {
+    "lancelot": {
+        "name": "Lancelot",
+        "power": 35,
+        "hp": 100,
+        "armour": [],
+        "weapon": {
+            "name": "Metal Sword",
+            "power": 50,
+        },
+        "potion": None,
+    },
+    "arthur": {
+        "name": "Arthur",
+        "power": 45,
+        "hp": 75,
+        "armour": [
+            {
+                "part": "helmet",
+                "protection": 15,
+            },
+            {
+                "part": "breastplate",
+                "protection": 20,
+            },
+            {
+                "part": "boots",
+                "protection": 10,
+            }
+        ],
+        "weapon": {
+            "name": "Two-handed Sword",
+            "power": 55,
+        },
+        "potion": None,
+    },
+    "mordred": {
+        "name": "Mordred",
+        "power": 30,
+        "hp": 90,
+        "armour": [
+            {
+                "part": "breastplate",
+                "protection": 15,
+            },
+            {
+                "part": "boots",
+                "protection": 10,
+            }
+        ],
+        "weapon": {
+            "name": "Poisoned Sword",
+            "power": 60,
+        },
+        "potion": {
+            "name": "Berserk",
+            "effect": {
+                "power": +15,
+                "hp": -5,
+                "protection": +10,
+            }
+        }
+    },
+    "red_knight": {
+        "name": "Red Knight",
+        "power": 40,
+        "hp": 70,
+        "armour": [
+            {
+                "part": "breastplate",
+                "protection": 25,
+            }
+        ],
+        "weapon": {
+            "name": "Sword",
+            "power": 45
+        },
+        "potion": {
+            "name": "Blessing",
+            "effect": {
+                "hp": +10,
+                "power": +5,
+            }
+        }
+    }
+}
+
+
+def dict_to_classes(dictionary: dict) -> list["Knight"]:
+    """
+    Transform incoming Config dictionary
+    into list of 4 elements -
+    each dictionary packed
+    into class "Knight"
+    """
+    knights = []
+    for knight, stats in dictionary.items():
+        armour = []
+        for element in stats["armour"]:
+            armour.append(Armour.from_element(element))
+        knight = Knight(
+            name=stats["name"],
+            power=stats["power"],
+            hp=stats["hp"],
+            armour=armour,
+            weapon=Weapon(
+                name=stats["weapon"]["name"],
+                power=stats["weapon"]["power"]
+            ),
+            potion=None if stats.get("potion") is None else Potion(
```

Comment:
Use second `get` paramter if potion is None

Reply:
potion must be None if it is not provided, so default value of None potion is None. Potion class works with None potion correct.

Reply:
I mean pass the Potion object into second paramter of get method, so you can avoid ternary here

Reply:
I don't understand, here is no ternay, here is None if potion is None or it is a Potion object. Passing  stats.get("potion", Potion(args) ) result in error in test. I need to pass None if potion is None.

Reply:
![image](https://github.com/mate-academy/py-knights-fighting/assets/106102798/03e36ca3-dead-4b9b-8d62-36d9bafe045c)